### PR TITLE
modify Ops to complex template

### PR DESCRIPTION
**Pull request by MingMingShangTian**
### PR types

Others
### PR changes

OPs
### Describe

Modify Ops to complex template, the Ops is as follow：

* conj
* real
* imag
* dot

### CODE REVIEW

File: paddle/fluid/operators/math/complex_functors.h [hunk 1]
```diff
@@ -66,7 +65,10 @@ using select_t = typename select<Head, Tail...>::type;
 template <typename T>
 using Real =
     select_t<cond<std::is_same<T, platform::complex64>::value, float>,
-             cond<std::is_same<T, platform::complex128>::value, double>, T>;
+             cond<std::is_same<T, platform::complex128>::value, double>,
```
Comment (chenwhql) [inline paddle/fluid/operators/math/complex_functors.h:68]: why complex64 and 128 are still needed here?

File: paddle/fluid/operators/math/complex_functors.h [hunk 2]
```diff
@@ -76,14 +78,18 @@ template <typename T, typename RealT>
 using NoComplex = typename std::enable_if<std::is_same<T, RealT>::value>::type;
 
 template <typename T>
-using EnableComplex =
-    typename std::enable_if<std::is_same<T, platform::complex64>::value ||
-                            std::is_same<T, platform::complex128>::value>::type;
+using EnableComplex = typename std::enable_if<
+    std::is_same<T, platform::complex64>::value ||
```
Comment (chenwhql) [inline paddle/fluid/operators/math/complex_functors.h:82]: same above

File: paddle/fluid/operators/math/complex_functors.h (hunk 1 above)
Comment (MingMingShangTian) [reply paddle/fluid/operators/math/complex_functors.h:68]: There are still some other Ops need the Real and EnableComplex template definition. They will be removed after the others Ops finishing replacing the complex64/128 types.

File: paddle/fluid/operators/math/complex_functors.h (hunk 2 above)
Comment (MingMingShangTian) [reply paddle/fluid/operators/math/complex_functors.h:82]: see the above.

**Review by chenwhql — APPROVED** [review]
LGTM

**Pull request merged**